File: API/Entities/Message.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Message
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public Guid Id { get; set; }
        public int SenderId { get; set; }
        public string SenderUsername { get; set; }
        public User Sender { get; set; }
        public int RecipientId { get; set; }
        public string RecipientUsername { get; set; }
        public User Recipient { get; set; }
        public string Content { get; set; }
        public DateTime? MessageReadDate { get; set; }
        public DateTime MessageSentDate { get; set; } = DateTime.Now;

    }
}

File: client/src/app/models/message.ts
export interface Message {
    id: string;
    senderId: number;
    senderUsername: string;
    senderBrandId: number;
    senderModelId: number;
    recipientId: number;
    recipientUsername: string;
    recipientBrandId: number;
    recipientModelId: number;
    content: string;
    messageReadDate?: Date;
    messageSentDate: Date;
}